src/formatting.rs

```diff
@@ -331,6 +331,9 @@ pub(crate) struct ReportedErrors {
 
     /// Formatted code differs from existing code (--check only).
     pub(crate) has_diff: bool,
+
+    /// Formatted code missed something, like lost comments or extra trailing space
+    pub(crate) has_missing_errors: bool,
```

Comment:
I think we should bikeshed on the name here, as the way the name currently reads is confusing. This flag is indicator that there was something that rustfmt failed to be able to format successfully (almost always caused by the presence of a comment in a place rustfmt cannot support comments).

It's really more indicative that there were parts of the input code that rustfmt failed to format and reverted to the original formatting, so there are unformatted portions of the code in the output.

Reply:
I forgot to mention the name in the discussion, as I couldn't find a good name for both lost-comments and trailing-white-spaces errors.  What about `has_lost_comments_error`? `has_unformatted_original_code`?  Another option is to split it to `has_lost_comments_error` and `has_trailing_whitespaces_error`.

Reply:
i'd prefer something that references "failed" or "unformatted"

Reply:
Renamed `has_missing_errors` to `has_unformatted_original_code`.  Hope this is o.k.

Reply:
I don't care too much about the internal name, but do have some reservations about the public facing function given that the rustfmt lib is available for outside consumers (and we can't change it once we put it out there).

I think it's important to keep `errors` in the name, so would prefer either:

(a) has_unformatted_code_errors
(b) has_unformatted_errors
(c) skip an aggregate helper, and just track/expose whitespace vs. lost comments separately (`has_lost_comment_errors` and `has_trailing_whitespace_errors`).

Please be sure to update the function name(s) accordingly too